### PR TITLE
add rule grouping tests

### PR DESCRIPTION
**Pull request by inashivb**
Previous PR: https://github.com/OISF/suricata-verify/pull/1653

- Still needs joingr test
- adds a disjointed overlap test

### CODE REVIEW

File: tests/rule-grouping/rule-grouping-2/test.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        tcp.toserver[11].rulegroup.rules[0].sig_id: 12
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[12].port: 1024
+        tcp.toserver[12].port2: 65534
+        tcp.toserver[12].rulegroup.id: 12
+        tcp.toserver[12].rulegroup.rules[0].sig_id: 13
+  - filter:
```
Comment (victorjulien) [inline tests/rule-grouping/rule-grouping-2/test.yaml:114]: this group should be part of group 12

File: tests/rule-grouping/rule-grouping-3/test.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        tcp.toserver[23].rulegroup.rules[0].sig_id: 13
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[24].port: 63978
+        tcp.toserver[24].port2: 65534
+        tcp.toserver[24].rulegroup.id: 12
+        tcp.toserver[24].rulegroup.rules[0].sig_id: 13
+  - filter:
```
Comment (victorjulien) [inline tests/rule-grouping/rule-grouping-3/test.yaml:222]: same: should be merged with 12

File: tests/rule-grouping/rule-grouping-4/test.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      count: 1
+      match:
+        tcp.toserver[0].port: 7
+        tcp.toserver[0].port2: 7
+        tcp.toserver[0].rulegroup.id: 0
+        tcp.toserver[0].rulegroup.rules[0].sig_id: 1
+        tcp.toserver[0].rulegroup.rules[1].sig_id: 2
+        tcp.toserver[0].rulegroup.rules[2].sig_id: 3
+        tcp.toserver[0].rulegroup.rules[3].sig_id: 6
+  - filter:
```
Comment (victorjulien) [inline tests/rule-grouping/rule-grouping-4/test.yaml:21]: merge into 1

File: tests/rule-grouping/rule-grouping-4/test.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[2].port: 25
+        tcp.toserver[2].port2: 42
+        tcp.toserver[2].rulegroup.id: 1
+        tcp.toserver[2].rulegroup.rules[0].sig_id: 2
+        tcp.toserver[2].rulegroup.rules[1].sig_id: 3
+        tcp.toserver[2].rulegroup.rules[2].sig_id: 5
+        tcp.toserver[2].rulegroup.rules[3].sig_id: 6
```
Comment (victorjulien) [inline tests/rule-grouping/rule-grouping-4/test.yaml:42]: this is a bit odd. 6 is 7-25, so it includes 25. This group is about 25-42. Should there be a group 25-25 and then 26-42?

File: tests/rule-grouping/rule-grouping-5/test.yaml [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        tcp.toserver[0].rulegroup.rules[0].sig_id: 1
+        tcp.toserver[0].rulegroup.rules[1].sig_id: 2
+        tcp.toserver[0].rulegroup.rules[2].sig_id: 3
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[1].port: 50
+        tcp.toserver[1].port2: 50
+        tcp.toserver[1].rulegroup.id: 0
```
Comment (victorjulien) [inline tests/rule-grouping/rule-grouping-5/test.yaml:26]: rule group id not assigned? I see multiple with 0

File: tests/rule-grouping/rule-grouping-4/test.yaml (hunk 3 above)
Comment (inashivb) [reply tests/rule-grouping/rule-grouping-4/test.yaml:21]: This is on TODO (ref: https://github.com/OISF/suricata/pull/10470/files#diff-26baf730c742afac833c40d8c1927c2040dcf619b387647e60f2aa34b4a9d58dR1168) Working on this today. Turns out I don't need to wait for the later stage, I can do this right after having found overlaps.

File: tests/rule-grouping/rule-grouping-4/test.yaml (hunk 4 above)
Comment (inashivb) [reply tests/rule-grouping/rule-grouping-4/test.yaml:42]: Indeed. That's how it should have been as per the algo 🤔 so, there's a bug. Thank you very much for the careful review

**Pull request closed without merging**